kahuna/public/js/util/async.js

```diff
@@ -68,6 +68,10 @@ async.factory("apiPoll", [
       if (status === 'fulfilled') {
         return $q.resolve(value);
       }
+      // Something has gone wrong, so we can let the user know
+      if (n > 100) {
+        throw new Error('apiPoll failed - Please try again.');
```

Comment:
Does this cancel the promise?

Reply:
The promise has already been settled at this point.